src/gen.cpp

```diff
@@ -520,15 +522,26 @@ int generate_cmake(const char *path, bool root) {
         }
     }
 
-    if (!cmake.vcpkg.version.empty()) {
-        assert("pmm is required in fetch-content for vcpkg to work" && cmake.contents.count("pmm") != 0);
-        comment("Bootstrap vcpkg");
-        cmd("include")("${pmm_SOURCE_DIR}/pmm.cmake");
-        tsl::ordered_map<std::string, std::vector<std::string>> vcpkg_args;
-        vcpkg_args["REVISION"] = {cmake.vcpkg.version};
-        vcpkg_args["REQUIRES"] = cmake.vcpkg.packages;
-        auto vcpkg = std::make_pair("VCPKG", vcpkg_args);
-        cmd("pmm")(vcpkg).endl();
+    if (!cmake.vcpkg.packages.empty()) {
+        cmd("include")("FetchContent");
+        cmd("message")("STATUS", "Fetching vcpkg...");
+        cmd("FetchContent_Declare")("vcpkg", "URL", "https://github.com/microsoft/vcpkg/archive/refs/tags/2021.05.12.tar.gz");
```

Comment:
Probably it would be cleaner to allow:

```toml
[vcpkg]
version = "2021.05.12"
packages = ["..."]
```

And generate the URL from that, or alternatively allow:

```toml
[vcpkg]
url = "https://my-company/vcpkg/blah.tar.gz"
packages = ["..."]
```

That way cmkr doesn't lock the user into a specific vcpkg version.

Reply:
added a url field to Vcpkg.